### ISSUE TITLE
Vertically center text not working

### ISSUE THREAD

**Issue by olimorris** [alacritty/alacritty#4834]
References: #105 #174 #475 which were resolved by #317

Unfortunately, the issue of vertically centering text appears to be not be resolved by #317 after I installed Alacritty for the first time today. I've included images below:

**Alacritty**
![Alacritty](https://user-images.githubusercontent.com/9512444/108780241-066c4080-7560-11eb-83bf-90d4ce257300.png)

**iTerm**
![iTerm](https://user-images.githubusercontent.com/9512444/108780269-12580280-7560-11eb-98aa-a117deb6588c.png)

### Config file
```yaml
window:
  padding:
    x: 5
    y: 5


# Font configuration
font:

  # Point size
  size: 21.0

  # Offset is the extra space around each character. `offset.y` can be thought
  # of as modifying the line spacing, and `offset.x` as modifying the letter
  # spacing.
  offset:
    x: 0
    y: 10

  # Glyph offset determines the locations of the glyphs within their cells with
  # the default being at the bottom. Increasing `x` moves the glyph to the
  # right, increasing `y` moves the glyph upward.
  glyph_offset:
    x: 0
    y: 10

  # Thin stroke font rendering (macOS only)
  #
  # Thin strokes are suitable for retina displays, but for non-retina screens
  # it is recommended to set `use_thin_strokes` to `false`.
  use_thin_strokes: true
```

### System

OS: macOS 11.2.1 (20D74) (Big Sur)
Version: alacritty 0.7.2 (5ac8060)

### Logs

Font/Terminal size:
```
[2021-02-22 22:36:46.700897000] [DEBUG] [alacritty] Using environment locale: en_GB.UTF-8
[2021-02-22 22:36:46.701422000] [INFO ] [alacritty] Welcome to Alacritty
[2021-02-22 22:36:46.701459000] [INFO ] [alacritty] Configuration files loaded from:
                                                      "/Users/Oli/.config/alacritty/alacritty.yml"
[2021-02-22 22:36:46.772167000] [DEBUG] [alacritty] Estimated DPR: 1
[2021-02-22 22:36:46.772249000] [DEBUG] [alacritty] Estimated window size: None
[2021-02-22 22:36:46.772270000] [DEBUG] [alacritty] Estimated cell size: 12 x 36
[2021-02-22 22:36:46.829655000] [INFO ] [alacritty] Device pixel ratio: 1
[2021-02-22 22:36:46.833123000] [INFO ] [alacritty] Initializing glyph cache...
[2021-02-22 22:36:46.844766000] [INFO ] [alacritty] ... finished initializing glyph cache in 0.01150523s
[2021-02-22 22:36:46.844892000] [INFO ] [alacritty] Cell size: 12 x 36
[2021-02-22 22:36:46.844918000] [INFO ] [alacritty] Padding: 5 x 5
[2021-02-22 22:36:46.844935000] [INFO ] [alacritty] Width: 800, Height: 600
[2021-02-22 22:36:46.846377000] [INFO ] [alacritty] PTY dimensions: Line(16) x Column(65)
[2021-02-22 22:36:46.850216000] [INFO ] [alacritty] Initialisation complete
[2021-02-22 22:36:47.682462000] [DEBUG] [alacritty_terminal] [unhandled osc_dispatch]: [['1',],['~',],] at line 813
[2021-02-22 22:36:49.715738000] [INFO ] [alacritty] Goodbye
```

**Comment by chrisduerr** [comment]
Text is position exactly as the font suggests, with no other adjustments. If your font sucks, Alacritty allows overriding everything necessary to mess with that, though you should probably just change fonts at that point.

**Comment by olimorris** [comment]
Why close the issue and not ask for clarification on the font used as I haven't explicitly specified it? Hence Alacritty will use OS default.

This is a hurried issue closure...

**Comment by chrisduerr** [comment]
Because it doesn't matter what font it is.

**Comment by olimorris** [comment]
So why isn't the fact that everything is seemingly setup okay, yet this issue is arising, not an open issue?

**Comment by chrisduerr** [comment]
Nothing is okay. Your font is crap.

**Comment by olimorris** [comment]
What font should I try then? Because here's a shot with Menlo...
![Screen Shot 2021-02-23 at 08 42 58](https://user-images.githubusercontent.com/9512444/108819767-25e08900-75b3-11eb-95d6-4a9b5b86da46.png)

**Comment by nixpulvis** [comment]
> with offset and glyph_offset at 15

Looks about right, I guess. I'm not sure why you are setting glyph offset to 15 though.

**Comment by nixpulvis** [comment]
I just realized you have set your `offset` to 10. As I understand things, people either add an offset to 1) effectively increase line hight, or 2) fix a font's bad scaling. In either case you may often need to adjust the `glyph_offset` as well, to keep things positioned correctly. This is going to depend on your font.

Well designed fonts I've used don't generally require this, but I'm also not generally trying to add space to my line height.

**Comment by nixpulvis** [comment]
> with offset and glyph_offset at 15

Looks about right, I guess. I'm not sure why you are setting glyph offset to 15 though. I'm assuming if you remove the `glyph_offset` your font will go back to being placed correctly at the bottom.

Basically, not all fonts are designed the same. Some fonts are centered differently, so we provide `glyph_offset` to allow users to adjust this at a basic level. It should be noted that you may often want a negative `glyph_offset` for fonts which render too high by default.

You might understand better if you read what kinds of properties exist for a font's glyphs, for example in FreeType: https://www.freetype.org/freetype2/docs/glyphs/glyphs-3.html, https://www.freetype.org/freetype2/docs/reference/ft2-base_interface.html

**Comment by olimorris** [comment]
@nixpulvis massively appreciated and really helpful context. Transitioning from iTerm2 has been a breeze. It was literally just this minor visual annoyance. Thank you!

**Comment by chroth7** [comment]
minor comment, if anyone struggles with similar issues: in my case, any change to `offset` took immediate effect (right after saving the file within alacritty) - whereas any changes to `glyph_offset` required restarting alacritty. This is on Mac.

aaand it worked, all to 0, using "Hasklug Nerd Font". Beautiful.

**Comment by kchibisov** [comment]
@chroth7 the particular issue with `font.glyph_offset` will be addressed in #5948.

**Comment by hsheikhali1** [comment]
Thank you @chroth7 and @kchibisov and @olimorris for not allowing this to be closed because apparently "your font sucks".
